### ISSUE TITLE
internal compiler error

### ISSUE THREAD

**Issue by cheesycod** [rust-lang/rust-clippy#8565]
### Summary

https://pastebin.com/uJFaC3Fg

### Reproducer

I tried this code:

```rust
cargo clippy -- -W clippy::pedantic
```

I expected to see this happen:

clippy successfully runs

Instead, this happened:

https://pastebin.com/uJFaC3Fg

### Version

```text
~/FatesList ❯❯❯ rustc -Vv
rustc 1.61.0-nightly (58f11791a 2022-03-17)
binary: rustc
commit-hash: 58f11791af4f97572e7afd83f11cffe04bbbd12f
commit-date: 2022-03-17
host: x86_64-unknown-linux-gnu
release: 1.61.0-nightly
LLVM version: 14.0.0
~/FatesList ❯❯❯
```

### Additional Labels

@rustlabel +ice

**Comment by matthiaskrgr** [comment]
This is a known rustc bug, should be fixed in the latest nightly: https://github.com/rust-lang/rust/issues/94998
Closing
